pennylane/devices/legacy_facade.py

```diff
@@ -167,6 +169,12 @@ def __init__(self, device: "qml.devices.LegacyDevice"):
         self._device = device
         self.config_filepath = getattr(self._device, "config_filepath", None)
 
+        if self._device.shots is not None and self._device.shots != Shots():
```

Comment:
How would `self._device.shots` be a `Shots` object? I thought it was always int or None.